### PR TITLE
Add Card vote Feature

### PR DESCRIPTION
**Pull request by NicoP-S**
I added a card voting feature. #796 
Can someone please review and test it?

---
This change is [Reviewable](https://reviewable.io/reviews/wekan/wekan/2994)

### CODE REVIEW

File: client/components/cards/cardDetails.styl [hunk 1]
```diff
@@ -330,3 +330,11 @@ card-details-color(background, color...)
 
 .card-details-indigo
   card-details-color(#4b0082, #ffffff) //White text for better visibility
+
+.voted
+  opacity: .7
+.vote-title
+  display: flex;
```
Comment (xet7) [inline client/components/cards/cardDetails.styl:337]: Please remove `;` from end of this, stylus does not have those.

File: client/components/cards/cardDetails.styl [hunk 2]
```diff
@@ -330,3 +330,11 @@ card-details-color(background, color...)
...
+  justify-content: space-between;
```
Comment (xet7) [inline client/components/cards/cardDetails.styl:338]: Please remove `;` from end of this, stylus does not have those.

File: client/components/cards/cardDetails.styl [hunk 3]
```diff
@@ -330,3 +330,11 @@ card-details-color(background, color...)
...
+.vote-result
+  display: flex;
```
Comment (xet7) [inline client/components/cards/cardDetails.styl:340]: Please remove `;` from end of this, stylus does not have those.

File: models/cards.js [hunk 4]
```diff
@@ -304,6 +304,38 @@ Cards.attachSchema(
       optional: true,
       defaultValue: '',
     },
+    vote: {
```
Comment (xet7) [inline models/cards.js:307]: For these new database fields, please also add migrations to Cards table/collection to bottom of `wekan/server/migrations.js`.

**Comment by xet7** [comment]
Optional if you like, not required:
You could export Trello JSON, and look can you import votes from Trello to Wekan here:
https://github.com/wekan/wekan/blob/master/models/trelloCreator.js#L258

**Comment by xet7** [comment]
@NicoP-S 

I did add most of those changes myself, and merged your vote feature. Thanks!

I did not add import from Trello.

**Comment by FuXXz** [comment]
I am very grateful for such efforts and that you have added it. It is really very nice.
Also for the reason that I can not contribute much because i  have only HTML and CSS knowledge :(

But the first tests showed that the feature is missing something.

Colleagues argued very quickly who voted yes or no. That means trouble :)

So, without any information about who voted, this function is severely restricted

**Comment by xet7** [comment]
@FuXXz 

When I'm voting for president, my vote is not public. Similarly in Wekan, I don't know why it should be known who voted what.

**Comment by xet7** [comment]
@FuXXz 

Your colleagues should just stop arguing and accept the results of voting :)

**Comment by NicoP-S** [comment]
Maybe it would be good to add a checkbox if the vote is public or not.
This should not be changeable after the vote Haß started.

**Comment by xet7** [comment]
@NicoP-S 

Yes it would be nice if you have time to add it like this, default unchecked:
```
[_] Public
```

**Comment by FuXXz** [comment]
@NicoP-S  unbelievable good and fast

**Pull request merged**